### PR TITLE
Update to Service Skeleton 1.0

### PR DESCRIPTION
**Pull request by davidtaylorhq**
This is a major refactoring, and some interfaces have changed. It will be a breaking change for all consumers, hence the major version bump.

Major changes include:
  - The way you start the service has changed. You must now use ServiceSkeleton::Runner to start it.
  - The Frankenstein metrics interface has changed slightly. Labels must be specified at initialization, and provided via a keyword argument when observing/incrementing

Other changes include:
- A DISABLE_INOTIFY switch has also been added, to aid people developing on macOS, where inotify is not supported.

### CODE REVIEW

File: config_skeleton.gemspec [hunk 1]
```diff
@@ -16,9 +16,8 @@ Gem::Specification.new do |s|
   s.required_ruby_version = ">= 2.3.0"
 
   s.add_runtime_dependency 'diffy', '~> 3.0'
-  s.add_runtime_dependency 'frankenstein', '~> 1.0'
```
Comment (cvx) [inline config_skeleton.gemspec:19]: 🤔 `frankenstein` is still a direct dependency of this gem – why is this gone from the gemspec?

File: lib/config_skeleton.rb [hunk 2]
```diff
@@ -2,11 +2,17 @@
 require 'fileutils'
 require 'frankenstein'
 require 'logger'
-require 'rb-inotify'
 require 'service_skeleton'
 require 'tempfile'
 require 'digest/md5'
 
+begin
+  require 'rb-inotify' unless ENV["DISABLE_INOTIFY"] == "1"
```
Comment (cvx) [inline lib/config_skeleton.rb]: For consistency with other instances of `ENV["DISABLE_INOTIFY"]`:

```suggestion
  require 'rb-inotify' unless ENV["DISABLE_INOTIFY"]
```

File: config_skeleton.gemspec (hunk 1 above)
Comment (davidtaylorhq) [reply config_skeleton.gemspec:19]: 🤦 fixed, thanks!

File: lib/config_skeleton.rb (hunk 2 above)
Comment (davidtaylorhq) [reply lib/config_skeleton.rb]: Updated

**Review by cvx — APPROVED** [review]

File: config_skeleton.gemspec (hunk 1 above)
Comment (davidtaylorhq) [reply config_skeleton.gemspec:19]: Oh I remember why now. Frankenstein is a transitive dependency of service_skeleton. So there is no need to pin a specific version here (and, in fact, pinning 1.0 breaks things).

Comment (cvx) [reply config_skeleton.gemspec:19]: It's not just transitive dependency, `config_skeleton` also uses it directly. It should be declared. (can be added without specifying version)

**Pull request merged**